### DISCUSSION TITLE
Is there a widely agreed upon deno cache directory?

### DISCUSSION THREAD

**Discussion by fushihara** [denoland/deno#26094] [General]
The library we are currently building requires not only the typescript code but also an arbitrary binary file for the library to work.
Every time the library code is loaded using the top level function, it checks for the existence of the file, and if it is not there, it downloads it from a remote location. This is nothing to worry about.

But where should we place the remotely downloaded files?

We could make it mandatory to specify this when using the library, or we could download it directly into the project.
but I would like to cache them somewhere global.

https://github.com/denosaurs/plug
For example, in this library
C:\Users\USERNAME\AppData\Local\deno\plugplug\https\github.com\HASHHASHHASHHASH.dll The file is located at
C:\Users\USERNAME\AppData\Local\deno\ is the value of DENO_DIR, but plug places the folder with the name of the library as it is.

This location is a directory for deno to use in the system, so we do not want to imitate the library to set up a directory here as it pleases.

**Comment by BlackAsLight** [comment]
I think the entire contents of DENO_DIR can be deleted and Deno will just re-download the things it needs. So as long as you name the folder something unique in there, something Deno isn't likely to use itself, you should be fine I'd imagine.
